openedx/core/djangoapps/embargo/views.py

```diff
@@ -2,10 +2,46 @@
 
 from django.http import Http404
 from django.views.generic.base import View
+from edx_rest_framework_extensions.authentication import JwtAuthentication
+from opaque_keys.edx.keys import CourseKey
+from rest_framework import permissions
+from rest_framework.response import Response
+from rest_framework.views import APIView
 
 from edxmako.shortcuts import render_to_response
 
 from . import messages
+from .api import check_course_access
+from .permissions import CanCallCheckCourseAccessAPI
+
+
+class CheckCourseAccessView(APIView):
+    permission_classes = (permissions.IsAuthenticated, CanCallCheckCourseAccessAPI)
```

Comment:
You don't really need the custom permission. Just restrict the view to staff users, which the service users are.